src/types/data_serialization_types.rs

```diff
@@ -67,28 +67,28 @@ pub fn decode_decimal(bytes: Vec<u8>) -> Result<f32, io::Error> {
     }
     let scaled = try_i_from_bytes(lr[1].clone());
     if scaled.is_err() {
-        return Err(scaled.unwrap_err());
+        return Err(unscaled.unwrap_err());
```

Comment:
It should return an error which was raised during evaluation of `scaled`.

For unscaled we return it in 66-th line